src/hotspot/share/c1/c1_ValueMap.hpp

```diff
@@ -148,6 +148,9 @@ class ValueNumberingVisitor: public InstructionVisitor {
       kill_memory();
     } else {
       kill_field(x->field(), x->needs_patching());
+      if (x->enclosing_field() != NULL) {
+        kill_field(x->enclosing_field(), true);
```

Comment:
Why do you need `all_offsets == true`? The offset of the enclosing field should always be known, right?